src/ui_menu_nbgl.c

```diff
@@ -47,60 +66,38 @@ static void exit(void)
 	os_sched_exit(-1);
 }
 
-static bool settings_navigation_callback(uint8_t page, nbgl_pageContent_t* content)
+static void settings_control_callback(int token, uint8_t index, int page)
 {
-	if (page == 0) {
-		content->type = INFOS_LIST;
-		content->infosList.nbInfos = NB_INFO_FIELDS;
-		content->infosList.infoTypes = (const char**)infoTypes;
-		content->infosList.infoContents = (const char**)infoContents;
-	} else if (page == 1) {
-		switches[0].text = "Expert mode";
-		switches[0].subText = "Enable expert mode";
-		switches[0].token = SWITCH_APP_MODE_TOKEN;
-		switches[0].tuneId = TUNE_TAP_CASUAL;
-		switches[0].initState = app_mode_expert();
-
-		content->type = SWITCHES_LIST;
-		content->switchesList.nbSwitches = NB_SETTINGS_SWITCHES;
-		content->switchesList.switches = (nbgl_layoutSwitch_t*)switches;
-	} else {
-		return false;
-	}
-	return true;
-}
-
-static void settings_control_callback(int token, uint8_t index)
-{
-	UNUSED(index);
+	UNUSED(page);
 	switch (token) {
-	case SWITCH_APP_MODE_TOKEN:
-		app_mode_set_expert(index);
-		break;
+		case SWITCH_APP_MODE_TOKEN:
+			app_mode_set_expert(index);
+			switches[0].initState = app_mode_expert();
+			break;
 
-	default:
-		PRINTF("Should not happen !");
-		break;
+		default:
+			PRINTF("Should not happen !");
+			break;
 	}
 }
 
-static void ui_menu_settings(void)
-{
-	nbgl_useCaseSettings("Cardano settings", PAGE_START, NB_PAGE_SETTING,
-	                     IS_TOUCHABLE, ui_idle_flow, settings_navigation_callback,
-	                     settings_control_callback);
-}
-
 void ui_idle_flow(void)
 {
 	// We need to make sure the ui context is reset even if the app restarts
 	nbgl_reset_transaction_full_context();
-	nbgl_useCaseHome("Cardano", &C_cardano_64, NULL, true,
-	                 ui_menu_settings, exit);
+
+	switches[0].text = "Expert mode";
+	switches[0].subText = "Enable expert mode";
+	switches[0].token = SWITCH_APP_MODE_TOKEN;
+	switches[0].tuneId = TUNE_TAP_CASUAL;
+	switches[0].initState = app_mode_expert();
+
+	nbgl_useCaseHomeAndSettings("Cardano", &C_cardano_64, NULL, INIT_HOME_PAGE, &settingContents,
```

Comment:
APPNAME instead of "Cardano"

Reply:
Problem is APPNAME is `Cardano ADA`, not `Cardano`... Don't know why but ...

Reply:
true !